pkg/network/ovn_kubernetes.go

```diff
@@ -256,24 +295,44 @@ func bootstrapOVNConfig(kubeClient client.Client) (*bootstrap.OVNConfigBoostrapR
 
 	if err != nil {
 		if apierrors.IsNotFound(err) {
-			klog.Infof("Did not find gateway-mode-config. Using default OVN configuration: %+v", ovnConfigResult)
-			return ovnConfigResult, nil
+			klog.Infof("Did not find gateway-mode-config")
 		} else {
 			return nil, fmt.Errorf("Could not determine gateway mode: %w", err)
 		}
+	} else {
+		modeOverride := cm.Data["mode"]
+		_, disableSNATMultipleGWsOverride := cm.Data["disable-snat-multiple-gws"]
+		if modeOverride != OVN_SHARED_GW_MODE && modeOverride != OVN_LOCAL_GW_MODE {
+			klog.Warningf("gateway-mode-config does not match %q or %q, is: %q. Using default OVN configuration: %+v", OVN_LOCAL_GW_MODE, OVN_SHARED_GW_MODE, modeOverride, ovnConfigResult)
+			return ovnConfigResult, nil
+		}
+		ovnConfigResult.GatewayMode = modeOverride
+		if disableSNATMultipleGWsOverride {
+			ovnConfigResult.EnableEgressIP = false
+			ovnConfigResult.DisableSNATMultipleGWs = true
+		}
+		klog.Infof("Overriding OVN configuration to %+v", ovnConfigResult)
 	}
-	modeOverride := cm.Data["mode"]
-	_, disableSNATMultipleGWsOverride := cm.Data["disable-snat-multiple-gws"]
-	if modeOverride != OVN_SHARED_GW_MODE && modeOverride != OVN_LOCAL_GW_MODE {
-		klog.Warningf("gateway-mode-config does not match %q or %q, is: %q. Using default OVN configuration: %+v", OVN_LOCAL_GW_MODE, OVN_SHARED_GW_MODE, modeOverride, ovnConfigResult)
-		return ovnConfigResult, nil
-	}
-	ovnConfigResult.GatewayMode = modeOverride
-	if disableSNATMultipleGWsOverride {
-		ovnConfigResult.EnableEgressIP = false
-		ovnConfigResult.DisableSNATMultipleGWs = true
+
+	dmc := types.NamespacedName{Namespace: "openshift-network-operator", Name: "dpu-mode-config"}
+	err = kubeClient.Get(context.TODO(), dmc, cm)
+
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Infof("Did not find dpu-mode-config")
+		} else {
+			return nil, fmt.Errorf("Could not determine Node Mode: %w", err)
+		}
+	} else {
+		nodeModeOverride := cm.Data["mode"]
```

Comment:
I think this is supposed to be `nodeModeOverride := dmc.Data["mode"]` ? Else this value will be local or shared gateway mode :) and below condition will always be hit.
Note that we are removing the hidden gateway-config-mode this release cycle, so hit this code conflict as I was trying to fix my PR - I can fix this in my PR, but I am surprised none of the testing caught this?
We should expose this as a proper API once this feature is stable since we are going to be overloading with config-maps for every config otherwise if we follow this pattern.

Reply:
oops nvm ignore my comment, my eyes read the object name differently...

Reply:
@tssurya Thanks for the comments and review!
Yes, we followed the same pattern as modeOverride (local/shared gateway mode) for DPU mode.
We will consider the API change for DPU node mode once the bluefield-2 design get mature and stable.